library/src/scala/tasty/Reflection.scala

```diff
@@ -27,6 +27,15 @@ abstract class Reflection
   def typeOf[T: scala.quoted.Type]: Type =
     implicitly[scala.quoted.Type[T]].unseal.tpe
 
+
+  /** Whether the code type checks in the given context?
+   *
+   *  @param code The code to be type checked
+   *
+   *  The code should be a sequence of expressions or statements that may appear in a block.
+   */
+  def typeChecks(code: String)(implicit ctx: Context): Boolean = kernel.typeChecks(code)(ctx)
```

Comment:
I would place it under an object. 

```scala
object checking {
  /** ... */
  def typeChecks(code: String)(implicit ctx: Context): Boolean = kernel.checking_typeChecks(code)(ctx)
}
```

Reply:
@nicolasstucki Are there potentially more APIs that can go to `checking`? Otherwise, it seems it's better to make the name more generic, like `typing`?.